lib/ansible/modules/clustering/k8s/k8s_auth.py

```diff
@@ -0,0 +1,278 @@
+  password:
+    description:
+    - Provide a password for authenticating with the API server.
+  ssl_ca_cert:
+    description:
+    - "Path to a CA certificate file used to verify connection to the API server. The full certificate chain
+      must be provided to avoid certificate validation errors."
+    - Only required if the CA certificate is not already present in the standard system CA trust store.
+  verify_ssl:
+    description:
+    - "Whether or not to verify the API server's SSL certificates."
+    type: bool
+  api_key:
+    description:
+    - When C(state) is set to I(absent), this specifies the token to revoke.
+
+requirements:
+  - python >= 2.7
+  - urllib3
+  - requests
+  - requests-oauthlib
+  - six
+'''
+
+EXAMPLES = '''
+- block:
+  # It's good practice to store login credentials in a secure vault and not
+  # directly in playbooks.
+  - include_vars: k8s_passwords.yml
+
+  - name: Log in
+    k8s_auth:
+      host: https://k8s.example.com/
+      ssl_ca_cert: ca.pem
+      username: admin
+      password: "{{ k8s_admin_password }}"
+    register: k8s_auth_results
+
+  - name: Store auth facts for quick access
+    set_fact:
+      k8s_auth_params:
+        host: "{{ k8s_auth_results.k8s_auth.host }}"
+        ssl_ca_cert: "{{ k8s_auth_results.k8s_auth.ssl_ca_cert }}"
+        api_key: "{{ k8s_auth_results.k8s_auth.api_key }}"
+
+  # Previous task generated I(k8s_auth) fact, which you can then use
+  # in k8s modules like this:
+  - name: Get a list of all pods from any namespace
+    k8s_facts:
+      <<: *k8s_auth_params
+      kind: Pod
+    register: pod_list
+
+  always:
+    - name: Always log out (revoke the token)
+      k8s_auth:
+        state: absent
+        <<: *k8s_auth_params
+'''
+
+RETURN = '''
+k8s_auth:
+  description: Kubernetes authentication facts.
+  returned: success
+  type: complex
+  contains:
+    api_key:
+      description: Authentication token.
+      returned: success
+      type: str
+    host:
+      description: URL for accessing the API server.
+      returned: success
+      type: str
+    ssl_ca_cert:
+      description: Path to a CA certificate file used to verify connection to the API server.
+      returned: success
+      type: str
+    verify_ssl:
+      description: "Whether or not to verify the API server's SSL certificates."
+      returned: success
+      type: bool
+    username:
+      description: Username for authenticating with the API server.
+      returned: success
+      type: str
+'''
+
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+import traceback
+import requests
+from urllib3.util import make_headers
+from requests_oauthlib import OAuth2Session
+from six.moves.urllib_parse import urlparse, parse_qs, urlencode
+
+K8S_AUTH_ARG_SPEC = {
+    'state': {
+        'default': 'present',
+        'choices': ['present', 'absent'],
+    },
+    'host': {'required': True},
+    'username': {},
+    'password': {'no_log': True},
+    'ssl_ca_cert': {},
```

Comment:
Should we `no_log` this as well? I still need to verify it can't be the actual cert content (rather than a path).

Reply:
k8s/common.py says:

```
    'ssl_ca_cert': {
        'type': 'path',
    },

Reply:
Hmm, as I look at the code, I'm not sure that's even implemented…

Reply:
It passes it through to the Kubernetes client's Configuration object which should handle it, but I don't think this path has been tested thoroughly. I'll see if I can find some time to investigate it soon

Reply:
Ok. And btw, CA cert chains are just public keys, so there's no need to `no_log` them.

Reply:
No, you wanted `no_log` to make sure there isn't hundreds of lines of pem files in the output. Makes sense.